src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -28,16 +32,30 @@ public HttpHandlerResourceV3Provider()
 
             HttpHandlerResourceV3 curResource = null;
 
-            var clientHandler = CreateCredentialHandler(source.PackageSource);
+            if (source.PackageSource.IsHttp)
```

Comment:
It tries to handle it gracefully. It's up to the caller really. For example if a local package source requested an http message handler it would be very odd and the caller requesting it would need to be fixed.